### PR TITLE
Add first integration tests

### PR DESCRIPTION
**Pull request by phlogistonjohn**
Based on PR #39 please review that first.

These changes add the very first integration tests - tests that expect to be run with a real kubernetes instance and all that fun stuff.
However, these are just the very first integration tests... not a full framework. They expect to be run with a KUBECONFIG env var set and pointed at a k8s cluster that is all set up, including access to an AD instance that matches the hard coded values in the test. The operator must also already be running.

Basically, this is a translation to Go of hand rolled stuff I have been using to test and demo the previous PRs that added the new CRDs. The plan is to build around these tests adding support for bringing up a test env and deploying the operator, etc. This is just a the first steps.

### CODE REVIEW

File: tests/integration/smb_share_test.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		return "", err
+	}
+	return pod.Status.PodIP, nil
+}
+
+func (s *SmbShareSuite) TestPodsReady() {
+	s.Require().NoError(s.waitForPodReady())
+}
+
+func (s *SmbShareSuite) TestShareAcess() {
```
Comment (obnoxxx) [inline tests/integration/smb_share_test.go:91]:
```suggestion
func (s *SmbShareSuite) TestShareAccess() {
```

**Review by obnoxxx — CHANGES_REQUESTED** [review]
First commit message has ` ... toe the code.`

**Comment by obnoxxx** [comment]
Problem I'm facing when trying to run this in kubernetes:

```
$ make deploy 
controller-gen not found in PATH, checking in GOBIN (/Users/madam/go/bin)...
/Users/madam/go/bin/controller-gen "crd:trivialVersions=true,crdVersions=v1" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
kustomize not found in PATH, checking in GOBIN (/Users/madam/go/bin)...
cd config/manager && /Users/madam/go/bin/kustomize edit set image controller=quay.io/samba.org/samba-operator:latest
/Users/madam/go/bin/kustomize build config/default | kubectl apply -f -
namespace/samba-operator-system created
customresourcedefinition.apiextensions.k8s.io/smbsecurityconfigs.samba-operator.samba.org created
customresourcedefinition.apiextensions.k8s.io/smbshares.samba-operator.samba.org configured
role.rbac.authorization.k8s.io/samba-operator-leader-election-role created
clusterrole.rbac.authorization.k8s.io/samba-operator-manager-role created
clusterrole.rbac.authorization.k8s.io/samba-operator-proxy-role created
clusterrole.rbac.authorization.k8s.io/samba-operator-metrics-reader created
rolebinding.rbac.authorization.k8s.io/samba-operator-leader-election-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/samba-operator-manager-rolebinding created
clusterrolebinding.rbac.authorization.k8s.io/samba-operator-proxy-rolebinding created
service/samba-operator-controller-manager-metrics-service created
deployment.apps/samba-operator-controller-manager created
$ make install
controller-gen not found in PATH, checking in GOBIN (/Users/madam/go/bin)...
/Users/madam/go/bin/controller-gen "crd:trivialVersions=true,crdVersions=v1" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
kustomize not found in PATH, checking in GOBIN (/Users/madam/go/bin)...
/Users/madam/go/bin/kustomize build config/crd | kubectl apply -f -
customresourcedefinition.apiextensions.k8s.io/smbsecurityconfigs.samba-operator.samba.org configured
customresourcedefinition.apiextensions.k8s.io/smbshares.samba-operator.samba.org configured
$ KUBECONFIG=~/.kube/config ./tests/test.sh 
=== RUN   TestSmbShares
=== RUN   TestSmbShares/1
    smb_share_test.go:35: 
                Error Trace:    smb_share_test.go:35
                                                        suite.go:102
                                                        smb_share_test.go:118
                Error:          Received unexpected error:
                                create not allowed while custom resource definition is terminating
                Test:           TestSmbShares/1
=== RUN   TestSmbShares/2
    smb_share_test.go:35: 
                Error Trace:    smb_share_test.go:35
                                                        suite.go:102
                                                        smb_share_test.go:134
                Error:          Received unexpected error:
                                create not allowed while custom resource definition is terminating
                Test:           TestSmbShares/2
--- FAIL: TestSmbShares (0.56s)
    --- FAIL: TestSmbShares/1 (0.35s)
    --- FAIL: TestSmbShares/2 (0.21s)
FAIL
FAIL    github.com/samba-in-kubernetes/samba-operator/tests/integration 1.279s
FAIL
$ 
```

**Comment by obnoxxx** [comment]
Actually, investigated a bit more:

```
$ kubectl get smbshares.samba-operator.samba.org    
NAME      AGE        
tshare1   17h          
```

so this has been created. but see:

```
$ kubectl describe smbshares.samba-operator.samba.org tshare1
...
Spec:
  Browseable:       false
  Read Only:        false
  Security Config:  sharesec1
  Share Name:       My Share
  Storage:
    Pvc:
      Spec:
        Access Modes:
          ReadWriteOnce
        Resources:
          Requests:
            Storage:         1Gi
        Storage Class Name:  rook-ceph-block
Events:                      <none>
```

and this does not set up rook-ceph :)

File: tests/integration/smb_share_test.go (hunk 1 above)
Comment (phlogistonjohn) [reply tests/integration/smb_share_test.go]: Fixed.

**Comment by phlogistonjohn** [comment]
I removed the explicit use of rook-ceph storage classes and now the embedded pvc subsections do not specify any particular storage class. This means that the tests now rely on the k8s cluster having a default storage class.

**Comment by phlogistonjohn** [comment]
Please do keep trying to run this locally until we get it working on a different setup than mine. :-D

**Comment by phlogistonjohn** [comment]
I tried being much more structured and not running the operator locally, but using the deployment and now I run into namespace issues, as I feared.
I'm setting DNM on this and will continue to work on the patches to at least allow the tests to work when the operator is deployed outside of "default". Later, I think work on the operator itself will be needed.

**Comment by phlogistonjohn** [comment]
Should be much better now. A number of issues and hard coded values are removed. It defaults to using the same namespace used by `make deploy` but this can be changed by setting a new SMBOP_TEST_NAMESPACE env var.

**Review by obnoxxx — APPROVED** [review]
lgtm
